crates/node/core/src/args/network.rs

```diff
@@ -148,9 +148,29 @@ pub struct NetworkArgs {
     /// Max capacity of cache of hashes for transactions pending fetch.
     #[arg(long = "max-tx-pending-fetch", value_name = "COUNT", default_value_t = DEFAULT_MAX_CAPACITY_CACHE_PENDING_FETCH, verbatim_doc_comment)]
     pub max_capacity_cache_txns_pending_fetch: u32,
+
+    /// Name of network interface used to communicate with peers.
+    ///
+    /// If flags is set, but no value is passed, the default interface for docker `eth0` is tried.
+    #[cfg(not(target_os = "windows"))]
+    #[arg(long = "net-if", conflicts_with = "addr", value_name = "IF_NAME")]
+    pub net_if: Option<String>,
```

Comment:
do we really need this?

can this be integrated as additional variant of the nat arg?

Reply:
this will happen when we integrate it into `NatResolver`, but that will grow the scope of this pr https://github.com/paradigmxyz/reth/pull/10681#issuecomment-2332612917